synapse/storage/databases/main/media_repository.py

```diff
@@ -363,6 +408,35 @@ async def mark_local_media_as_safe(self, media_id: str, safe: bool = True) -> No
             desc="mark_local_media_as_safe",
         )
 
+    async def count_pending_media(self, user_id: UserID) -> Tuple[int, int]:
+        """Count the number of pending media for a user.
+        Returns:
+            A tuple of two integers: the total pending media requests and the earliest
+            expiration timestamp.
+        """
+
+        def get_pending_media_txn(txn: LoggingTransaction) -> Tuple[int, int]:
+            sql = (
+                "SELECT COUNT(*), MIN(created_at)"
+                " FROM local_media_repository"
+                " WHERE user_id = ?"
+                " AND created_at > ?"
+                " AND media_length IS NULL"
+            )
```

Comment:
I usually find these clearer as multi-line strings:

```suggestion
            sql = """
            SELECT COUNT(*), MIN(created_at)
            FROM local_media_repository
            WHERE user_id = ?
                AND created_at > ?
                AND media_length IS NULL
            """
```